File: src/main/java/com/efedotov/vaultly/dto/file/FileChunkDto.java
package com.efedotov.vaultly.dto.file;

import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class FileChunkDto {
    private String fileName;
    private String contentType;
    private byte[] data;
    private boolean lastChunk;
    private UUID folderId;
}

File: frontend/lib/server/model/file/file_chunk_dto/file_chunk_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'dart:typed_data';

import 'package:vaulth_app/server/model/file/file_upload_request/uint8_list_converter.dart';

part 'file_chunk_dto.freezed.dart';
part 'file_chunk_dto.g.dart';

@freezed
abstract class FileChunkDto with _$FileChunkDto {
  const factory FileChunkDto({
    required String fileName,
    required String contentType,
    @Uint8ListConverter() Uint8List? data,
    required bool lastChunk,
    String? folderId,
  }) = _FileChunkDto;

  factory FileChunkDto.fromJson(Map<String, dynamic> json) =>
      _$FileChunkDtoFromJson(json);
}
